### PR TITLE
Stabilize cfg_target_has_atomic

### LINKED ISSUE #32976: Tracking issue for #[cfg(target_has_atomic = ...)]
Tracking https://github.com/rust-lang/rfcs/pull/1543

### PR DESCRIPTION
**Pull request by Amanieu**
`target_has_atomic_equal_alignment` is now tracked separately in #93822.

Partially touches on #32976 (maybe closes, but nominated to check on load_store being part of the stabilization)

### CODE REVIEW

**Review by davidtwco — APPROVED** [review]

**Comment by davidtwco** [comment]
Looks like you've already had an FCP for this in #32976 so I'll just go ahead and above.

@bors r+

File: compiler/rustc_feature/src/builtin_attrs.rs [hunk 1]
```diff
@@ -26,12 +26,10 @@ const GATED_CFGS: &[GatedCfg] = &[
     // (name in cfg, feature, function to check if the feature is enabled)
     (sym::target_abi, sym::cfg_target_abi, cfg_fn!(cfg_target_abi)),
     (sym::target_thread_local, sym::cfg_target_thread_local, cfg_fn!(cfg_target_thread_local)),
-    (sym::target_has_atomic, sym::cfg_target_has_atomic, cfg_fn!(cfg_target_has_atomic)),
-    (sym::target_has_atomic_load_store, sym::cfg_target_has_atomic, cfg_fn!(cfg_target_has_atomic)),
```
Comment (Mark-Simulacrum) [inline compiler/rustc_feature/src/builtin_attrs.rs:30]: Is it intentional that we're stabilizing load_store as well? My recollection of T-lang discussion was that we were intending to limit to just target_has_atomic on the first pass.

**Comment by Amanieu** [comment]
I wasn't present in the lang meeting and assumed `target_has_atomic_load_store` would be stabilized as well since I haven't seen any comments saying otherwise.

**Comment by Mark-Simulacrum** [comment]
OK -- I'm not sure either. I guess this is another point in favor of clear FCPs on long-lived tracking issues with little in the description. Let's not r- this to avoid canceling the rollup as well, and I can make sure that @rust-lang/lang puts this on its agenda on Tuesday next week to discuss whether to stabilize has_atomic_load_store as well.

**Pull request merged**